### PR TITLE
Fix copy paste mistake

### PR DESCRIPTION
**Pull request by Suchiman**
Fix copy paste mistake in `RequestHeaderEncodingSelector` (from `ResponseHeaderEncodingSelector`)

### CODE REVIEW

File: xml/System.Net.Http/SocketsHttpHandler.xml [hunk 1]
```diff
@@ -724,8 +724,8 @@ For example, if the value is 64, then 65,536 bytes are allowed for the maximum r
         <ReturnType>System.Net.Http.HeaderEncodingSelector&lt;System.Net.Http.HttpRequestMessage&gt;</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>Gets or sets a callback that decodes request headers values.</summary>
-        <value>The header encoding selector callback to decode the value for the specified response header name, or <see langword="null" /> to indicate the default behavior.</value>
+        <summary>Gets or sets a callback that encodes request headers values.</summary>
```
Comment (antonfirsov) [inline xml/System.Net.Http/SocketsHttpHandler.xml:727]: The whole phrasing seems to be very inaccurate. I would rather write:
```suggestion
        <summary>Gets or sets a callback that selects the <see cref="System.Text.Encoding"/> to encode header values.</summary>
```

**Review by antonfirsov — COMMENTED** [review]
I think we should improve the phrasing even more. /CC @MihaZupan

File: xml/System.Net.Http/SocketsHttpHandler.xml (hunk 1 above)
Comment (antonfirsov) [reply xml/System.Net.Http/SocketsHttpHandler.xml:727]: This text has to be also fixed in `ResponseHeaderEncodingSelector` .

**Review by scalablecory — APPROVED** [review]
Thanks!

**Pull request merged**